toolkit/retry_decorator.py

```diff
@@ -0,0 +1,19 @@
+from time import sleep
+
+def retry(exceptions, tries, delay=2, backoff=2):
+
+    def decorator(func):
+
+        def wrapper(*args,**kwargs):
+            nonlocal delay
+            for i in range(tries):                
+
```

Comment:
Would like to see some kind of printing functionalities here for informing about the state of the process, like mentioning how many times did the function try